src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -763,7 +779,8 @@ def update_connectedk8s(cmd, instance, tags=None):
 
 
 def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy="", http_proxy="", no_proxy="", proxy_cert="",
-                  disable_proxy=False, kube_config=None, kube_context=None, auto_upgrade=None):
+                  disable_proxy=False, kube_config=None, kube_context=None, auto_upgrade=None, container_registry_repository=None,
+                  container_registry_username=None, container_registry_password=None, default_registry=None, anonymous_pull=None):
```

Comment:
One more thing. For moving from mcr to pcr, it would make sense to add a --update-timeout parameter as well.(just like we do for connect and upgrade commands).

Reply:
It is already there in the onboarding i.e. connect command. If we want to take the same functionality for update commands as well we will create a separate PR because it will affect both PCR and MCR